app/assets/javascripts/toggle_name_and_email_input.js

```diff
@@ -7,16 +7,20 @@
         var link = $($(this).data('link'));
         var targetScope = $(this).data('target');
         var store = target.detach();
-        var form = $('form').last();
+        var contentForm = $('form:last');
```

Comment:
WRT `closest()`, you may have had a scoping issue.  Try moving the `contentForm` variable assignment outside of the click event.

```javascript
this.each(function(){
  var contentForm = $(this).closest('form');
  $(this).on('click', function(){
  ...
```